libtenzir/builtins/operators/from_to_2.cpp

```diff
@@ -0,0 +1,714 @@
+      }
+      auto file_ending = std::string_view{file}.substr(first_dot);
+      // determine compression based on ending
+      auto compression_extension = std::string_view{};
+      std::tie(compression_extension, compression_name)
+        = determine_compression(file_ending);
+      if (not compression_name.empty()) {
+        for (const auto& p : plugins::get<operator_factory_plugin>()) {
+          const auto name = p->name();
+          // TODO These should ultimately be different operators
+          if (name == "decompress") {
+            decompress_plugin = p;
+            break;
+          }
+        }
+        TENZIR_ASSERT(decompress_plugin);
+      }
+      // determine read operator based on file ending
+      std::tie(read_plugin, read_properties) = find_formatter_given(
+        file_ending, &operator_factory_plugin::read_properties,
+        &operator_factory_plugin::read_properties_t::extensions, path,
+        compression_extension, inv.args.front().get_location(), docs, ctx);
+    }
+  post_deduction:
+    TENZIR_TRACE("from operator: given pipeline size   : {}",
+                 pipeline_argument
+                   ? static_cast<int>(pipeline_argument->inner.body.size())
+                   : -1);
+    TENZIR_TRACE("from operator: determined loader     : {}",
+                 load_plugin ? load_plugin->name() : "none");
+    TENZIR_TRACE("from operator: loader accepts pipe   : {}",
+                 load_plugin ? load_properties.accepts_pipeline : false);
+    TENZIR_TRACE("from operator: loader produces events: {}",
+                 load_plugin ? load_properties.events : false);
+    TENZIR_TRACE("from operator: determined decompress : {}",
+                 decompress_plugin ? decompress_plugin->name() : "none");
+    TENZIR_TRACE("from operator: determined read       : {}",
+                 read_plugin ? read_plugin->name() : "none");
+    /// TODO: Decide on whether/where we actually want this
+    if (not read_plugin and not has_pipeline_or_events
+        and load_properties.default_format) {
+      read_plugin = load_properties.default_format;
+      TENZIR_TRACE("from operator: fallback read         : {}",
+                   read_plugin->name());
+      diagnostic::warning(
+        "The deduced load operator `{}` suggests `{}`, which will be used.",
+        strip_prefix(load_plugin->name()), strip_prefix(read_plugin->name()))
+        .emit(ctx);
+    }
+    if (not load_plugin) {
+      return failure::promise();
+    }
+    if (not read_plugin and not has_pipeline_or_events) {
+      return failure::promise();
+    }
+    if (not has_pipeline_or_events) {
+      inv.args.emplace_back(ast::pipeline_expr{});
+      pipeline_argument = try_as<ast::pipeline_expr>(inv.args.back());
+      if (decompress_plugin) {
+        auto decompress_ent = ast::entity{std::vector{
+          ast::identifier{strip_prefix(decompress_plugin->name()),
+                          location::unknown},
+        }};
```

Comment:
This is somewhat brittle and won't work with modules etc. I wonder whether there is a better way to do this, given that we already know the plugin and the arguments that we want to use… We could ship the compiled pipeline to something like `load_tcp`, for example by allowing compiled pipelines as constants. But maybe that's not worth it right now.

Reply:
I'll just resolve this. We won't address this as part of this PR.